TACC/Core-Portal
# bug/WP-40  Prevent push keys message and modal for key service systems

## Overview

The goal for this task is to prevent showing a Push Keys message for systems that use a key service. For key service systems, pushing additional keys can cause them to break. For non key service systems the functionality needed to remain as is. 

## Related

* [WP-40](https://jira.tacc.utexas.edu/browse/WP-40)

## Changes

- Added a check in AppsForm and DataFilesTable components that checks whether or not a system is a key service system and in case of an error, different error messages are shown based on the system. For a key service system the user is prompted to submit a ticket and for a non key service system the user is prompted to push their keys. 
- Added support for default values for the fetchSelectedSystem hook since certain features were breaking without this. 

## Testing

1. Via command line, delete your corral credentials, replacing `$USERNAME` with your username:
  a. `client.systems.removeUserCredential(systemId='cloud.data', userName=$USERNAME)`
2. Wait for cache to clear in Tapis 
3. Check My Data (Corral) in Data Files and confirm updated error message with 'submit a ticket' option is displayed
4. Open an application from the Applications section and confirm updated error message with 'submit a ticket' option is displayed. 
5. Repeat the above process this time removing credentials from the **frontera** system (which is a non key service system) and confirm that the error message in My Data (Frontera) gives the option to 'push your keys'

## UI

- Default error message for non key service systems 

<img width="1919" alt="Screen Shot 2023-07-17 at 1 13 05 PM" src="https://github.com/TACC/Core-Portal/assets/23081932/93a78b52-bdc2-4fa6-903a-6769150e7fcb">



- Updated error messages for key service systems

<img width="1919" alt="Screen Shot 2023-07-17 at 1 07 02 PM" src="https://github.com/TACC/Core-Portal/assets/23081932/fabc1d1c-cce4-4ff9-bdf4-32661bd86f92">

<img width="1919" alt="Screen Shot 2023-07-17 at 1 07 18 PM" src="https://github.com/TACC/Core-Portal/assets/23081932/1324b853-40f7-4ce1-8b45-2ba8751bc1a5">


## Notes
